### DISCUSSION TITLE
How to post an image?

### DISCUSSION THREAD

**Discussion by timmyers** [bluesky-social/atproto#1077] [How to do it?] [answered]
I'm having some trouble figuring out how to use this library to upload an image post.  Are here any examples of this?

**Comment by ianklatzco** [comment, marked as answer]
https://github.com/ianklatzco/atprototools#two-liner-to-get-started-your-second-time

see "post an image"

if you'd like the raw http it's a 'createBlob' or 'uploadBlob' followed by 'createRecord' (where the record key contains an embed key with the identifier returned by the blob upload)

**Reply by bnewbold** [reply]
In addition to Ian's example, this blog post describes uploading and referencing images in a post: https://atproto.com/blog/create-post
